app/src/main/java/com/arctouch/codechallenge/home/HomeFragment.java

```diff
@@ -0,0 +1,160 @@
+package com.arctouch.codechallenge.home;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ProgressBar;
+import android.widget.SearchView;
+
+import com.arctouch.codechallenge.R;
+import com.arctouch.codechallenge.controller.HomeController;
+import com.arctouch.codechallenge.data.Cache;
+import com.arctouch.codechallenge.details.DetailsFragment;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.util.EndlessRecyclerViewScrollListener;
+
+import java.util.List;
+
+/**
+ * Created by wanderleyfilho on 7/22/18.
+ */
+
+public class HomeFragment extends Fragment {
+
+    private RecyclerView recyclerView;
+    private ProgressBar progressBar;
+    private HomeController homeController;
+    private long currentPage = 1L;
+    private List<Movie> movieList;
+    private EndlessRecyclerViewScrollListener endlessRecyclerViewScrollListener;
+    private boolean isSearching = false;
+    private String querySearch;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view =  inflater.inflate(R.layout.home_fragment, container, false);
+        setHasOptionsMenu(true);
+
+        this.recyclerView = view.findViewById(R.id.recyclerView);
+        this.progressBar = view.findViewById(R.id.progressBar);
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getActivity());
+        recyclerView.setLayoutManager(linearLayoutManager);
+        endlessRecyclerViewScrollListener = new EndlessRecyclerViewScrollListener(linearLayoutManager) {
+            @Override
+            public void onLoadMore(int page, int totalItemsCount, RecyclerView view) {
+                currentPage += 1L;
+                getNextMovies();
+            }
+        };
+        this.recyclerView.addOnScrollListener(endlessRecyclerViewScrollListener);
+
+        homeController = new HomeController();
+        homeController.loadGenres().subscribe(response -> {
+            Cache.setGenres(response.genres);
+            loadMovies();
+        });
+
+        return view;
+    }
+
+    public void loadMovies() {
+        homeController.loadUpcomingMoviesByPage(this.currentPage).subscribe(response -> {
```

Comment:
You are subscribing to the controller to be able to retrieve the movies. Can you think of a way to remove these callbacks from your view?

Reply:
I could write it differently putting a function as a success parameter for that subscriber response, the code would appear more friendly. Right now I can't think other way of removing the callbacks, as it is an asynchronous method, the view should be listening in a background thread for that response and mount it on the UI once it is done, without freezing the screen, so the user can interact with the other elements which are not dependent on that job.